builtin/providers/aws/resource_aws_spot_fleet_request.go

```diff
@@ -377,10 +384,15 @@ func buildSpotFleetLaunchSpecification(d map[string]interface{}, meta interface{
 	}
 
 	if v, ok := d["key_name"]; ok {
-		opts.KeyName = aws.String(v.(string))
+		key_name := v.(string)
+		err := validateSpotFleetRequestKeyName(key_name)
```

Comment:
Can you add this as the "validateFunc" in the schema section? I think that would be more ideomatic
https://godoc.org/github.com/hashicorp/terraform/helper/schema#SchemaValidateFunc